source/slurm/overview.rst

```diff
@@ -332,6 +332,169 @@ allowed.
+number of tasks on more nodes. The Memory High Water mark information
+will help you determine how many nodes you would have to use to satisfy the
+memory requirements of your job. There are a couple of different ways of
+getting the memory usage information about your job.
+
+Using report-mem utility in batch jobs
+--------------------------------------
+
+To get the maximum amount of memory (also called "Memory High Water mark") used
+up to a specific point in your job, you can add the following command to your
+job file:
+
+   ``report-mem``
+
+Typically, the best place to put this command is the end of your job
+file or altered exit points if your jobs are written such that they may exit
+before the end.
+
+There may be instances where this is not feasible because you
+don't have direct access to the job file. For example, you might be using other
+scripts to generate job files on the fly, where users have the option to
+specify launch option in a "config" file. In those instances, you can get a
+memory report for your parallel jobs using ``--epilog`` option of the srun
+command as shown below:
+
+.. code-block:: shell
+
+   srun --epilog=/apps/local/bin/report-mem   wrf.exe
+
+Using report-mem utility on a job that is currently running
+-----------------------------------------------------------
+
+If your job is currently running on the system and you would like to find the
+Memory High Water Mark up to that point, use
+the ``report-mem`` command from a
+login node on that job, as in the example below:
+
+.. code-block:: shell
+
+    hfe03.% report-mem -j 4665051
+    Peak memory usage summary:
+    min = 11139788 KB
+    ave = 11181442 KB
+    max = 11261556 KB
+    All nodes sorted by peak memory as percentage of limit: (in KB)
+    % of user user user total total
+    Node limit max limit current current phys
+    h16c50 12.0 11261556 94208000 11259356 14455952 97609020
+    h25c22 11.9 11208488 94208000 11207184 14486172 97609020
+    h25c17 11.9 11178112 94208000 11177692 14508136 97609024
+    h25c40 11.8 11152296 94208000 11151424 14451696 97609024
+    h25c48 11.8 11148416 94208000 11147668 14445588 97609024
+    h25c20 11.8 11139788 94208000 11139672 14465660 97609024
+    hfe03.%
+
+Determining the amount of memory used by a process
+--------------------------------------------------
+
+The techniques above give you the amount of memory used on each node,
+rather than the amount of memory used by each task.
+
+To find the amount of memory used by each task, use this method:
+
+    Submit the job, but use a full node (using ``sbatch -N 1 ...`` for example)
+
+If your execute line is:
+
+   ``./myexe``
+
+replace it with
+
+   ``/usr/bin/time ./myexe``
+
+If you search for the string "elapsed", you will find a line resembling the
+following:
+
+.. code-block:: shell
+
+   1.34user 15.57system 0:22.76elapsed 74%CPU (0avgtext+0avgdata 7822876 maxresident)k
+
+
+which shows that this process used approximately 7.8 GB of memory.
```

Comment:
Maybe explain how we arrive at this calculation?   (7,822,876 * 1k = 7,822,876,000 bytes ~ 7.8GB)